warehouse/templates/manage/token.html

```diff
@@ -0,0 +1,96 @@
+{#
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ # http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+-#}
+{% extends "manage_base.html" %}
+
+{% set user = request.user %}
+{% set title = "API token management" %}
+
+{% block account_active %}vertical-tabs__tab--is-active{% endblock %}
+{% block account_mobile_active %}vertical-tabs__tab--is-active{% endblock %}
+
+{% block title %}{{ title }}{% endblock %}
+
+{% block main %}
+  <h1 class="page-title">{{ title }}</h1>
+
+  {% if serialized_macaroon %}
+  <section id="provisioned-key">
+    <div class="callout-block callout-block--warning">
+      <p>
+        This token will only appear once. Copy it now.
+      </p>
+      <!-- TODO: This needs to be word-wrapped -->
+      <p>
+        {{ serialized_macaroon }}
+      </p>
+    </div>
+  </section>
+  {% endif %}
+
+  {% if user.macaroons|length > 0 %}
+  <section id="active-tokens">
+    <h2>Active API tokens</h2>
+    <table class="table table--light">
+      <thead>
+        <th>Description</th>
+        <th>Identifier</th>
+        <th>Created</th>
+        <th colspan="2">Last used</th>
+      </thead>
+      <tbody>
+        {% for macaroon in user.macaroons %}
+        <tr>
+          <td>{{ macaroon.description }}</td>
+          <td>{{ macaroon.id }}</td>
+          <td>{{ humanize(macaroon.created) }}</td>
+          <td>{{ humanize(macaroon.last_used) if macaroon.last_used else "Never" }}</td>
+          <td class="table__action">
+            <a href="#delete-token" class="button button--primary">Delete</a>
+            {% set title="Remove API token - " + macaroon.description %}
+            {% set confirm_button_label="Remove API token" %}
+            {% set action=request.route_path('manage.account.token.delete') %}
+            {% set extra_fields %}
+              <input type="text" name="macaroon_id" value="{{ macaroon.id }}" hidden>
+            {% endset %}
+            {{ confirm_modal(title=title, confirm_name="Username", confirm_string=user.username, confirm_button_label=confirm_button_label, slug="delete-token", extra_fields=extra_fields, action=action, warning=False, confirm_string_in_title=False) }}
+          </td>
+        </tr>
+        {% endfor %}
+      </tbody>
+    </table>
```

Comment:
![wordwrap-issue-api-keys](https://user-images.githubusercontent.com/842790/61420471-a49e1c00-a8d0-11e9-956f-903b987cc1db.png)

Two items:
* should we allow the user to make a second key with an identical description to an existing key? I think we shouldn't, to help the user avoid confusion.
* word wrap issue

Reply:
> * should we allow the user to make a second key with an identical description to an existing key? I think we shouldn't, to help the user avoid confusion.

I agree; this was part of the original schema that I adopted but it makes sense to enforce uniqueness here to me.



> * word wrap issue

Yep; I might need @nlhkabu's help fixing this one 😅